File: .claude-cpp/instructions.cpp
// instructions.cpp  --  your message to Claude, written in C++.
//
//   1. Edit this file.
//   2. Save (Cmd+S). It is compiled and run; the panel on the right shows the
//      text it produced.
//   3. When the preview looks right, press "Send to Claude"
//      (or Cmd+Option+Enter).
//
// Compiler errors show up here as red squiggles and in the panel. Fixing them
// is part of the job -- that is the point of this tool.

#include "claude.hpp"

#include <string>
#include <vector>

using namespace claude;

int main() {
    Prompt p;

    p.goal("Explain what this project does and how it is organised.");

    // Point Claude at the parts of the project that matter. Each of these
    // warns (in the panel) if the path does not exist.
    p.context({File("README.md"), Dir("src")});

    // Ordinary C++ builds the steps: a vector and a range-based for loop.
    const std::vector<std::string> steps = {
        "Read the README and skim the top-level files.",
        "List the main components and what each is responsible for.",
        "Summarise the whole thing in three bullet points.",
    };
    for (const std::string& s : steps) {
        p.step(s);
    }

    p.constrain("Do not modify any files.");
    p.accept("I can explain the project to someone else after reading your answer.");

    return p.emit();
}

File: templates/instructions.py
# instructions.py  --  your message to Claude, written in Python.
#
#   1. Edit this file.
#   2. Save (Cmd+S). It is run; the panel on the right shows the text it
#      produced.
#   3. When the preview looks right, press "Send to Claude"
#      (or Cmd+Option+Enter).
#
# You do not write English here. You combine concepts -- Subjects (things),
# Actions (verbs), Conditions (done-when) -- and claude.py words them.
# Your own concepts for this project go in project.py.
#
# Errors show up here as red squiggles and in the panel. Fixing them is part
# of the job -- that is the point of this tool.

from project import *

p = Prompt()

p.goal(Explain(Purpose(this_project), Structure(this_project)))

# Ordinary Python builds the steps: a list and a for loop.
# Each of these warns (in the panel) if the path does not exist.
to_read = [File("README.md"), Dir("src")]
for thing in to_read:
    p.step(Read(thing))
p.step(List(Components(this_project)))
p.step(Summarize(this_project).in_(Bullets(3)))

p.forbid(Modify(this_project))

p.emit()
